utils/params.go

```diff
@@ -791,28 +792,54 @@ func readConfigFromTarget(client vcsclient.VcsClient, gitParamsFromEnv *Git) (co
 	return
 }
 
-// This function fetches a config profile if JF_CONFIG_PROFILE is provided.
-// If so - it verifies there is only a single module with a '.' path from root. If these conditions doesn't hold we return an error.
-func getConfigProfileIfExistsAndValid(xrayVersion, xscVersion string, jfrogServer *coreconfig.ServerDetails) (configProfile *services.ConfigProfile, err error) {
+// This function attempts to fetch a config profile if JF_USE_CONFIG_PROFILE is set to true.
+// If we need to use a profile, we first try to get the profile by name that can be provided through JF_CONFIG_PROFILE. If name is provided but profile doesn't exist we return an error.
+// If we need to use a profile, but name is not provided, we check if there is a config profile associated to the repo URL.
+// When a profile is found we verify several conditions on it.
+// If a profile was requested but not found by url nor by name we return an error.
+func getConfigProfileIfExistsAndValid(xrayVersion, xscVersion string, jfrogServer *coreconfig.ServerDetails, gitClient vcsclient.VcsClient, gitParams *Git) (configProfile *services.ConfigProfile, err error) {
+	useConfigProfile := strings.ToLower(getTrimmedEnv(JfrogUseConfigProfileEnv))
+	if useConfigProfile == "false" || useConfigProfile == "" {
+		log.Debug(fmt.Sprintf("Configuration Profile usage is disabled. All configurations will be derived from environment variables and files.\nTo enable a Configuration Profile, please set %s to TRUE", JfrogUseConfigProfileEnv))
+		return
+	}
+
+	// Attempt to get the config profile by profile's name
 	profileName := getTrimmedEnv(JfrogConfigProfileEnv)
-	if profileName == "" {
-		log.Debug(fmt.Sprintf("No %s environment variable was provided. All configurations will be induced from Env vars and files", JfrogConfigProfileEnv))
+	if profileName != "" {
+		log.Debug(fmt.Sprintf("Configuration profile was requested. Searching profile by provided name '%s'", profileName))
```

Comment:
do we need a log if we create error on that just few lines later? can't this info be included in the err?

Reply:
Yes because this log isn't intended to indicate about an error, but rather to add information about the way we are going to fetch the Config Profile if requested. 
We have a similar log before GetConfigProfileByUrl that indicates fetch using URL